static/js/tools/stat_var/dataset_selector.tsx

```diff
@@ -0,0 +1,231 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component to select a source and dataset.
+ */
+
+import axios from "axios";
+import React, { useEffect, useState } from "react";
+import { Card, Container, CustomInput } from "reactstrap";
+
+import { NamedPlace } from "../../shared/types";
+
+const PREFIX = "dataset-selector";
+const MAX_SUGGESTIONS = 5;
+
+interface DatasetSelectorProps {
+  // Filter StatVarHierarchy to stat vars that exist for the entities.
+  filterStatVars: (entities: NamedPlace[]) => void;
+  // Map of source name to dcid.
+  sourceMap: Record<string, string>;
+}
+
+function simplify(s: string): string {
+  return s.replace(/[^0-9a-z]/gi, "").toLowerCase();
+}
+
+export function DatasetSelector(props: DatasetSelectorProps): JSX.Element {
+  const [activeSuggestion, setActiveSuggestion] = useState(0);
+  const [datasets, setDatasets] = useState([]);
+  const [filteredSuggestions, setFilteredSuggestions] = useState(
+    Object.keys(props.sourceMap)
+  );
+  const [showSuggestions, setShowSuggestions] = useState(false);
+  const [suggestionsWidth, setSuggestionsWidth] = useState(null);
+  const [userInput, setUserInput] = useState("");
+
+  const handleResize = () => {
+    const width = document.getElementById(`${PREFIX}-ac`).offsetWidth;
+    if (width !== suggestionsWidth) {
+      setSuggestionsWidth(width);
+    }
+  };
+
+  useEffect(() => {
+    handleResize();
+    window.addEventListener("resize", handleResize);
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  });
+
+  async function updateDatasets(source: string): Promise<void> {
+    if (!(source in props.sourceMap)) {
+      return;
+    }
+    const dcid = props.sourceMap[source];
+    const datasetsPromise = await axios.get(
+      `/api/browser/propvals/isPartOf/${dcid}`
+    );
+    const currentDatasets = [];
+    for (const dataset of datasetsPromise?.data?.values?.in) {
```

Comment:
curious why doing `await` and `datasetsPromise?` instead of the usual 
```
datasetsPromise = axios.get(...)
datasetsPromise.then(...).catch(...)
```

Reply:
changed to use this pattern instead